### PR TITLE
issue-112: added @RestController to the SharedObjects PMD rules

### PR DESCRIPTION
**Pull request by stokpop**
This pull request is for issue 112.

Also updated javadoc and added unit tests for @RestController for the SharedObjects PMD rules.

### CODE REVIEW

File: src/test/resources/com/jpinpoint/perf/lang/java/ruleset/concurrent/xml/AvoidUnguardedAssignmentToNonFinalFieldsInSharedObjects.xml [hunk 1]
```diff
@@ -477,4 +477,19 @@ public class AdviesEnAfsluitkostenFunction extends AbstractFunction<AdviesEnAfsl
...
+            <description>Avoid unguarded assignment to non-final fields in shared objects: @RestController</description>
+            <expected-problems>1</expected-problems>
+            <expected-linenumbers>6</expected-linenumbers>
+            <code><![CDATA[
+@RestController
+class ARestController1 {
+    private String url; //
+
+    public void setUrlBad(final String url) {
+        this.url = url;
```
Comment (jborgers) [inline src/test/resources/com/jpinpoint/perf/lang/java/ruleset/concurrent/xml/AvoidUnguardedAssignmentToNonFinalFieldsInSharedObjects.xml:490]: pls put '// bad' at the actual line of the violation

File: rulesets/java/jpinpoint-rules.xml [hunk 2]
```diff
@@ -1,5 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<ruleset name="jpinpoint-rules" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 http://pmd.sourceforge.net/ruleset_2_0_0.xsd" xmlns="http://pmd.sourceforge.net/ruleset/2.0.0" xmlns:fn="http://www.w3.org/TR/xpath-functions/">
```
Comment (jborgers) [inline rulesets/java/jpinpoint-rules.xml:2]: Removed is: xmlns:fn="http://www.w3.org/TR/xpath-functions/"
For what reason?

**Review by jborgers — CHANGES_REQUESTED** [review]
All good, except 1 inline question and 1 inline improvement request. 
One more question: are all occurrences of the singleton annotations in the rules now extended with @RestController?

File: rulesets/java/jpinpoint-rules.xml (hunk 2 above)
Comment (stokpop) [reply rulesets/java/jpinpoint-rules.xml:2]: idea says: not used, but I see it gets generated with the merge call: put the original merged file in commit

File: src/test/resources/com/jpinpoint/perf/lang/java/ruleset/concurrent/xml/AvoidUnguardedAssignmentToNonFinalFieldsInSharedObjects.xml (hunk 1 above)
Comment (stokpop) [reply src/test/resources/com/jpinpoint/perf/lang/java/ruleset/concurrent/xml/AvoidUnguardedAssignmentToNonFinalFieldsInSharedObjects.xml]: fixed

File: rulesets/java/jpinpoint-rules.xml (hunk 2 above)
Comment (stokpop) [reply rulesets/java/jpinpoint-rules.xml]: ah, it was gone again: it is the "optimize imports" option that I have enabled in intellij: it removes unused imports at commit automatically!!

Comment (stokpop) [reply rulesets/java/jpinpoint-rules.xml]: "One more question: are all occurrences of the singleton annotations in the rules now extended with @RestController?"

I think so, I looked for all places in merged file where @Controller is used, let me know if you see an additional one.

**Comment by stokpop** [comment]
_"One more question: are all occurrences of the singleton annotations in the rules now extended with @RestController?"_

I think so, I looked for all places in merged file where @controller is used, let me know if you see an additional one.

**Review by jborgers — APPROVED** [review]

**Pull request merged**